lib/atom-io-client.coffee

```diff
@@ -187,3 +187,34 @@ class AtomIoClient
 
   getCachePath: ->
     @cachePath ?= path.join(remote.app.getPath('userData'), 'Cache', 'settings-view')
+
+  search: (query, options) ->
+    qs = {q: query}
+
+    if options.themes
+      qs.filter = 'theme'
+    else if options.packages
+      qs.filter = 'package'
+
+    if options.sort
+      qs.sort = options.sort
+
+    options = {
+      url: "#{@baseURL}packages/search"
+      headers: {'User-Agent': navigator.userAgent}
+      qs: qs
+      json: true
+    }
+
+    new Promise (resolve, reject) ->
+      request options, (err, res, body) ->
```

Comment:
I'm unfamiliar with proxying but is there anything special that needs to be done here to avoid proxy issues?

Reply:
```coffeescript
    if @httpsProxy?
      options.proxy = @httpsProxy
    if @strictSsl is false
      options.rejectUnauthorized = false
```

is what is needed to be added to the request options when that pull request is merged.